map_test.go

```diff
@@ -1910,38 +1998,73 @@ func BenchmarkMarshaling(b *testing.B) {
 }
 
 func BenchmarkPerCPUMarshalling(b *testing.B) {
+	batch := 1024
 	newMap := func(valueSize uint32) *Map {
 		m, err := NewMap(&MapSpec{
 			Type:       PerCPUHash,
 			KeySize:    8,
 			ValueSize:  valueSize,
-			MaxEntries: 1,
+			MaxEntries: uint32(batch),
 		})
 		if err != nil {
 			b.Fatal(err)
 		}
 		return m
 	}
 
-	key := uint64(1)
-	val := []uint64{1, 2, 3, 4, 5, 6, 7, 8}
+	possibleCPUs, err := PossibleCPU()
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	val := make([]uint64, possibleCPUs)
+	for i := range val {
+		val[i] = uint64(i + 1)
+	}
 
 	m := newMap(8)
-	if err := m.Put(key, val[0:]); err != nil {
-		b.Fatal(err)
+	for i := 0; i < batch; i++ {
+		key := uint64(i + 1)
```

Comment:
I know this is in the original code as well, but why do we start at `1`?

Reply:
I like to ensure non-zero values for testing.